canopen/pdo/base.py

```diff
@@ -268,7 +268,7 @@ def on_message(self, can_id, data, timestamp):
             with self.receive_condition:
                 self.is_received = True
                 self.data = data
-                self.period = timestamp - self.timestamp
+                self.period = timestamp - self.timestamp if self.timestamp is not None else timestamp
```

Comment:
Maybe `self.period` should remain None until two messages have been received.

```python
if self.timestamp is not None:
    self.period = timestamp - self.timestamp
```

Reply:
Yes, that makes sense. :)